util/progress/progress.go

```diff
@@ -274,3 +274,20 @@ func (pw *noOpWriter) Write(_ string, _ interface{}) error {
 func (pw *noOpWriter) Close() error {
 	return nil
 }
+
+func OneOff(ctx context.Context, id string) func(err error) error {
```

Comment:
Just now notice that; guess it's because I'm not familiar with the terminology, but `id` is confusing if it means "any text to print", or is the `id` meant to be unique? (so printing the same message twice will discard one of them?)

Reply:
@jedevc just mentioned to me on slack that it's indeed an `id` (and "bad things can happen if you use the same one twice").

Wondering if (for the nature of these to be "just an arbitrary string to print) the actual ID should be generated internally (separate from the actual text), e.g., could be a GUID that's generated?